graylog2-server/src/main/java/org/graylog/mcp/server/MarkdownBuilder.java

```diff
@@ -353,12 +461,29 @@ public MarkdownBuilder lineBreak() {
     }
 
     public MarkdownBuilder raw(String content) {
-        sb.append(content);
+        sb.append(escapeMarkdownInstance(content, MdContext.PARAGRAPH));
+        return this;
+    }
+
+    /**
+     * Appends raw content without any markdown escaping.
+     * Use this method when you want to insert literal Markdown syntax or
+     * when you have already escaped the content yourself.
+     * <p>
+     * <b>Warning:</b> Using this method with untrusted user input may allow
+     * markdown injection and break document structure.
+     *
+     * @param content The raw content to append (no escaping applied)
+     * @return This MarkdownBuilder instance for method chaining
+     */
+    public MarkdownBuilder unsafeRaw(String content) {
+        sb.append(orEmpty(content));
         return this;
     }
 
     @Override
     public String toString() {
+        if (this.isCodeBlockOpen) closeCodeBlock();
```

Comment:
style: we never leave out `{` and `}` and never keep them in a single line for consistency and clarity